content/blog/posts/o3de-graphics-atom-intro.md

```diff
@@ -0,0 +1,35 @@
+From there, I started writing my own material type. I referenced the shader code in a `.azsl` file via entry points in a `.shader` file, which itself was referenced in the overarching `.materialtype` file. This gave me a broad view of how Atom renders each material and also taught me the function of each shader resource group, function, pass, and more. While the number of files and connections were intimidating at first, once I became familiar with the terms and what each component did, I appreciated Atom’s pipeline organization and code modularization.
+
+An interesting aspect of graphics that I noticed is the focus on optimization. While many fields of computer science promote optimization, graphics optimization appears particularly important because a low frame rate is obvious. Also, since graphics rely heavily on vector math and physics, optimizing graphics code isn’t necessarily limited to choosing which data structure or algorithm to use, but also which mathematical model to deploy. I encountered a simple example when I needed to use an oscillation function to mimic vegetation bending. The most obvious function is the sine wave. However, I learned that using sine is considered relatively slow, so it is more ideal to use a sine lookup table or to use triangle waves.
+
+## Looking back
+
+{{< image-width src="/images/blog/graphics-atom-intro/tree.png" width="100%" alt="The progress of making a tree." >}}
+
+As I added more shader code and more passes, I saw my material evolving, eventually looking and swaying like a real-life tree! That, to me, was one of the most rewarding parts of using O3DE and writing shader code: the almost immediate visual feedback. Additionally, looking back at the process, I appreciate the beauty of how everything fits together. My material type uses multiple passes, with each pass editing a particular part of the material via my custom shaders.
+
+{{< image-width src="/images/blog/graphics-atom-intro/tree-sway.gif" width="100%" alt="GIF of trees swaying." >}}
+
+I'm thankful for this internship at Amazon, and for the experience to be part of the O3DE community and contribute to Atom while learning graphics. One of Amazon's leadership principles that resonates with me is [customer obsession](https://www.amazon.jobs/en/principles). This principle is reflected in the way Atom is built. Its shaders generally aren't written to be end-all be-all, but instead, are openly customizable. Because Atom is part of this real-time 3D engine, those who use the engine may want to make their own material types or write their own custom shaders. The material types included in O3DE seem to be common features that game developers may want, such as skin. While I was still new to graphics, working with these general-purpose materials was helpful since they served as great templates. In turn, the point of my projects for my internship is to create and write tutorials to allow users to learn and create custom material types, just as I did.
```

Comment:
```suggestion
I'm thankful for this internship at Amazon, and for the experience to be part of the O3DE community and contribute to Atom while learning graphics. One of Amazon's leadership principles that resonates with me is [customer obsession](https://www.amazon.jobs/en/principles). This principle is reflected in the way Atom is built. Its shaders generally aren't written to be end-all be-all, but instead, are openly customizable. Because Atom is part of this real-time 3D engine, those who use the engine may want to make their own material types or write their own custom shaders. The material types included in O3DE seem to be common features that game developers may want, such as skin. While I was still new to graphics, working with these general-purpose materials was helpful since they served as great templates. In turn, the point of my projects for my internship this summer is to create and write tutorials to allow users to learn and create custom material types, just as I did.
```
Great blog post Judy! It inspires me to take a hack at making my own shader.  Minor suggestion, _only_ if you like it: ... the point of my projects for my internship [this year _or_ summer] is to create and ...

Reply:
Thanks, glad to hear! And, yes, adding "this summer" is a good suggestion.